presto-main/src/main/java/com/facebook/presto/sql/planner/GroupingOperationRewriter.java

```diff
@@ -54,13 +54,16 @@ public static Expression rewriteGroupingOperation(GroupingOperation expression,
         }
         else {
             checkState(groupIdVariable.isPresent(), "groupId symbol is missing");
+            List<FieldId> fieldIdList = columnReferenceFields.get(NodeRef.of(expression.getGroupingColumns().get(0)));
+            checkState(fieldIdList.size() == 1, "grouping column must reference exactly one field");
 
-            RelationId relationId = columnReferenceFields.get(NodeRef.of(expression.getGroupingColumns().get(0))).getRelationId();
+            RelationId relationId = fieldIdList.get(0).getRelationId();
```

Comment:
refactor this into a getter/helper function

Reply:
Done